lucille-core/src/test/java/com/kmwllc/lucille/core/RunnerTest.java

```diff
@@ -787,4 +787,10 @@ public void testMetrics() throws Exception {
     // 1) other metrics collected by Stage beyond the counts in the processDocumentTime timer
     // 2) metrics collected by Indexer, Worker, Publisher
   }
+
+  @Test
+  public void testRenderFlag() throws Exception {
```

Comment:
suggest adding an additional test for the case where the config has a substitution that can't be resolved
check that Runner.renderConfig throws ConfigException (which is a kind of RuntimeException)
this new test would use a different config that references an environment variable named something like LUCILLE_TEST_CONFIG_RENDER_UNDEFINED